lib/os/heap-validate.c

```diff
@@ -386,3 +386,66 @@ void sys_heap_print_info(struct sys_heap *heap, bool dump_chunks)
 {
 	heap_print_info(heap->heap, dump_chunks);
 }
+
+#ifdef CONFIG_SYS_HEAP_RUNTIME_STATS
+
+int heap_runtime_stats_get(struct sys_heap *heap,
+		struct sys_heap_runtime_stats *stats)
+{
+	size_t free_bytes = 0;
+	struct z_heap *h;
+	chunkid_t c = 0;
+
+	if ((heap == NULL) || (stats == NULL)) {
+		return -EINVAL;
+	}
+
+	h = heap->heap;
+	stats->free_bytes = 0;
+	stats->allocated_bytes = 0;
+	stats->max_avail_bytes = 0;
+
+	/* Iterate through all chunks */
+	do {
+		if (chunk_used(h, c)) {
+			if ((c != 0) && (c != h->end_chunk)) {
+				stats->allocated_bytes += chunksz_to_bytes(
+						h, chunk_size(h, c));
+			} else {
+				;
+			}
+		} else {
+			if (!solo_free_header(h, c)) {
+				free_bytes = chunksz_to_bytes(h,
+						chunk_size(h, c));
+				stats->free_bytes += free_bytes;
+				if (free_bytes > stats->max_avail_bytes) {
+					stats->max_avail_bytes = free_bytes;
+				} else {
+					;
+				}
+			} else {
+				;
+			}
+		}
+		c = right_chunk(h, c);
+	} while (c != h->end_chunk);
+
+	return 0;
+}
+
+#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
```

Comment:
No point in putting an extern function inside an ifdef.  It won't be linked if it's not referenced.